pkg/forwarder/forwarder.go

```diff
@@ -657,15 +563,15 @@ func (f *DefaultForwarder) SubmitOrchestratorChecks(payload Payloads, extra http
 	return f.submitProcessLikePayload(orchestratorEndpoint, payload, extra, true)
 }
 
-func (f *DefaultForwarder) submitProcessLikePayload(ep endpoint, payload Payloads, extra http.Header, retryable bool) (chan Response, error) {
+func (f *DefaultForwarder) submitProcessLikePayload(ep transaction.Endpoint, payload Payloads, extra http.Header, retryable bool) (chan Response, error) {
```

Comment:
I could be wrong but are all of these function signature changes using by-val copies of the params vs by-ref? If so, that seems like a pretty big overhead in their invocation.

Reply:
>I could be wrong but are all of these function signature changes using by-val copies of the params vs by-ref

Why do you think there is a change between by-val vs by-ref? `type endpoint` was renamed to `type Endpoint` and moved into `transaction` package (Endpoint is not an interface).

Reply:
The comment here wasn't about your specific changes to it but the actual implementation of them - by passing in `ep` as a `struct`, they're being copied before being passed on to the method which seems like it could have some overhead in hot-path code. It's not something that needs to be fixed as part of the PR but it does seem like it could benefit from being passed by-ref.